### PR TITLE
Add upgrade command

### PR DESCRIPTION
**Pull request by wangxiyuan**
This pr add openlab ha upgrade tool and doc

### CODE REVIEW

File: deploy.py [hunk 1]
```diff
@@ -57,6 +70,20 @@ def add_cli_args():
     return parser
 
 
+def list_changes(project, days):
```
Comment (Yikun) [inline deploy.py:73]: nit: list-change

File: playbooks/upgrade-ha-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      args:
+        executable: /bin/bash
+
+- name: Install and config openlabcmd
+  hosts: zuul-master, zuul-slave
+  become: yes
+  tasks:
+    - name: upgrade zuul
+      shell: |
+        openlab ha node set `hostname` --maintain yes
```
Comment (Yikun) [inline playbooks/upgrade-ha-deployment.yaml]: Should we also reset the maintain status back to `no` after upgrade ?

File: playbooks/upgrade-ha-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+---
+- name: Install and config openlabcmd
+  hosts: nodepool-master, nodepool-slave
+  become: yes
+  tasks:
+    - name: upgrade nodepool
+      shell: |
```
Comment (Yikun) [inline playbooks/upgrade-ha-deployment.yaml:7]: Should we consider to set the `set -e` to enable the error exit to do short exit when error happen.

File: ha_healthchecker/README.md [hunk 4]
```diff
@@ -193,3 +193,15 @@ If you want to mange the cluster, we now provide two actions:
...
+## Upgrade
+
+We usually upgrade OpenLab environment once a month to keep the zuul and nodepool as new as possible. The upgrade work rely one labkeeper tool. Here is the upgrade step:
+
+1. command `./deploy.py openlab-ha list-change`
+    This command will print the newly merged patches during last 31 days from zuul and nodepool community. Operator should check these changes first. If any change will break current deployment, operator should fix it by hand.
+2. Once the check is passed, command `./deploy.py openlab-ha upgrade`
+    This command will upgrade the zuul and nodepool source code to the newest, re-generate zuul-web page and restart related services. During this action, zuul and nodepoool nodes will be set to `maintaining` status.
+3. Once `upgarde` action is done, operator should check the deployment to ensure every service works well. If not, operator should fix it by hand.
+4. Once all services works as expect, command `./deploy.py openlab-ha upgrade-complete`. This command will set all nodes back to `up` status.
```
Comment (Yikun) [inline ha_healthchecker/README.md]: nit: Once all services **work** as expect

File: ha_healthchecker/README.md [hunk 5]
```diff
@@ -193,3 +193,15 @@ If you want to mange the cluster, we now provide two actions:
...
+
+## Upgrade
+
+We usually upgrade OpenLab environment once a month to keep the zuul and nodepool as new as possible. The upgrade work rely one labkeeper tool. Here is the upgrade step:
+
+1. command `./deploy.py openlab-ha list-change`
+    This command will print the newly merged patches during last 31 days from zuul and nodepool community. Operator should check these changes first. If any change will break current deployment, operator should fix it by hand.
+2. Once the check is passed, command `./deploy.py openlab-ha upgrade`
+    This command will upgrade the zuul and nodepool source code to the newest, re-generate zuul-web page and restart related services. During this action, zuul and nodepoool nodes will be set to `maintaining` status.
+3. Once `upgarde` action is done, operator should check the deployment to ensure every service works well. If not, operator should fix it by hand.
```
Comment (Yikun) [inline ha_healthchecker/README.md]: it seems not set back to `up` in implementation

**Review by Yikun — CHANGES_REQUESTED** [review]

File: ha_healthchecker/README.md (hunk 5 above)
Comment (Yikun) [inline ha_healthchecker/README.md]: upgrade

File: deploy.py (hunk 1 above)
Comment (wangxiyuan) [reply deploy.py:73]: function name should use _

Comment (Yikun) [reply deploy.py:73]: I mean list_change, but I see you call the list_changes under list_change arg. So, I don't think it's necessary change.

[1] https://github.com/theopenlab/labkeeper/blob/acb1d02/deploy.py#L112

**Review by Yikun — APPROVED** [review]

**Review by bzhaoopenstack — APPROVED** [review]
Go for test

**Pull request merged**